### PR TITLE
feat(ST-98): get and create message api endpoints

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- added messages module, controller, service
- added POST `messages` api endpoints and necessary function in the service layer
- added GET `messages/classrooms/:id` api endpoints and necessary function in the service layer

## Associated Ticket Link(s)

- https://trello.com/c/p1RA1yRz

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/messages/messages.controller.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { CreateMessageDto } from "./messages.dto";
+import { MessagesService } from "./messages.service";
+
+UseInterceptors(ResponseTransformInterceptor);
+UseGuards(JwtAuthGuard);
+@Controller("messages")
+export class MessagesController {
+  constructor(private readonly messagesService: MessagesService) {}
+
+  @Get("classrooms/:id")
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.controller.ts:24]: Please follow Rest principles.
Should be `classrooms/:classroomsId/messages`

File: src/messages/messages.controller.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  getMessages(@Param("id", ParseIntPipe) id: number) {
+    return this.messagesService.getMessages(id);
+  }
+
+  @Post()
+  addMessage(@Body() createMessageDto: CreateMessageDto) {
+    console.log("Hellp");
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.controller.ts:31]: Let's remove unnecessary console logs

File: src/messages/messages.controller.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@Controller("messages")
+export class MessagesController {
+  constructor(private readonly messagesService: MessagesService) {}
+
+  @Get("classrooms/:id")
+  getMessages(@Param("id", ParseIntPipe) id: number) {
+    return this.messagesService.getMessages(id);
+  }
+
+  @Post()
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.controller.ts:29]: Same comment as above. Let's follow REST principles

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending Ishmam's comments

File: src/classrooms/classrooms.controller.ts [hunk 4]
```diff
@@ -77,4 +81,14 @@ export class ClassroomsController {
...
+
+  @Get(":id/messages")
+  getMessages(@Param("id", ParseIntPipe) id: number) {
+    return this.messagesService.getMessages(id);
+  }
+
+  @Post("messages")
+  addMessage(@Body() createMessageDto: CreateMessageDto) {
+    return this.messagesService.addMessage(createMessageDto);
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.controller.ts:93]: You don't have to move this to classrooms controller. The endpoint should be the same but in messages controller.

File: src/classrooms/classrooms.controller.ts [hunk 5]
```diff
@@ -77,4 +81,14 @@ export class ClassroomsController {
 
     return this.classroomsSerializer.serializeMany(students);
   }
+
+  @Get(":id/messages")
+  getMessages(@Param("id", ParseIntPipe) id: number) {
+    return this.messagesService.getMessages(id);
+  }
+
+  @Post("messages")
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.controller.ts:90]:
```suggestion
  @Post(":id/messages")
```

File: src/messages/messages.service.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    await this.em.persistAndFlush(message);
+
+    return message;
+  }
+
+  async getMessages(classroomId: number) {
+    const messages = await this.messagesRepository.find({ classroom: classroomId });
+
+    return messages;
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.service.ts:26]: In both cases check if the classroom exists with findOneOrFail

File: src/messages/messages.controller.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  constructor(private readonly messagesService: MessagesService) {}
+
+  @Get(":id/messages")
+  getMessages(@Param("id", ParseIntPipe) id: number) {
+    return this.messagesService.getMessages(id);
+  }
+
+  @Post(":id/messages")
+  addMessage(@Param("id", ParseIntPipe) id: number, @Body() createMessageDto: CreateMessageDto) {
+    return this.messagesService.addMessage(createMessageDto);
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.controller.ts:17]: You probably need to pass the classroomId as well

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**